File: codingame/onboarding.java
import java.util.*;
class Player{
	public static void main(String[]args){
		Scanner cin=new Scanner(System.in);
		for(;;){
			String enemy1 = cin.next(); // name of enemy 1
			int dist1 = cin.nextInt(); // distance to enemy 1
			String enemy2 = cin.next(); // name of enemy 2
			int dist2 = cin.nextInt(); // distance to enemy 2

			System.out.println(dist1<dist2 ? enemy1 : enemy2);
		}
	}
}

File: codingame/onboarding.kt
import java.util.*
fun main(args : Array<String>) {
	val input = Scanner(System.`in`)

	// game loop
	while (true) {
		val enemy1 = input.next() // name of enemy 1
		val dist1 = input.nextInt() // distance to enemy 1
		val enemy2 = input.next() // name of enemy 2
		val dist2 = input.nextInt() // distance to enemy 2

		println(if(dist1<dist2) enemy1 else enemy2)
	}
}
